### PR TITLE
ROCm-5.1.3 toolchain and low-level runtimes against llvm/clang-14

### PR DESCRIPTION
**Pull request by littlewu2508**

### CODE REVIEW

**Comment by littlewu2508** [comment]
@heroxbd I think it is time to merge ROCm 5.1.3 toolchains.

Before I was hesitating due to two problems:
1. the hardcoded clang include dir in rocm-comgr and hip. It will cause bugs if clang minor version bumped without rebuild these two packages. It's still not resolved, but it won't cause any problem now, because the llvm upstream decided that there will be no patch version for llvm/clang-14, so the minor version number will stay at 14.0.6. I already find a potential solution to this (probably a rocm-comgr bug) and contact reported the issue in https://github.com/RadeonOpenCompute/ROCm-CompilerSupport/issues/48. In the future (>5.2, with clang-15) this shall come to a solution.
2. gcc-12 issue. This should leave to each affected ebuild, because masking gcc-12 in dev-util/hip does not work. Although this is related to hip, building hip itself with gcc-12 have no issue. Ebuilds should pick the suitable compiler at setup time.

So in conclusion, the low-level runtimes and toolchains for ROCm-5.1.3 is suitable to be merged now.

**Comment by heroxbd** [comment]
as a dependence of `hip`, should `roctracer` be bumped together?

**Comment by heroxbd** [comment]
> as a dependence of `hip`, should `roctracer` be bumped together?

If you are merging `roctracer` and `rccl` into `hip`, add blocking dependence to `hip`, like `!dev-util/roctracer`.

**Comment by heroxbd** [comment]
If this bump won't break `rocBLAS` and friends, it is time to relax their SLOT dependence on `hip` to `:=`.

Otherwise, you should unkeyword all the new packages till `rocBLAS-5.1.3` is in tree because at present they cannot satisfy the `DEPEND` and `RDEPEND` of `rocBLAS`, etc.

**Comment by littlewu2508** [comment]
On Wed, Aug 03, 2022 at 06:35:36AM -0700, Benda Xu wrote:
> as a dependence of `hip`, should `roctracer` be bumped together?
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/gentoo/gentoo/pull/26441#issuecomment-1203959860
> You are receiving this because you were mentioned.
> 
> Message ID: ***@***.***>
dev-util/hip-5.1.3 does not depend on roctracer. It just use roctracer's
header so it's just downloaded as another distfile
--

**Comment by littlewu2508** [comment]
On Wed, Aug 03, 2022 at 06:53:59AM -0700, Benda Xu wrote:
> If this bump won't break `rocBLAS` and friends, it is time to relax their SLOT dependence on `hip` to `:=`.
> 
> Otherwise, you should unkeyword all the new packages till `rocBLAS-5.1.3` is in tree because at present they cannot satisfy the `DEPEND` and `RDEPEND` of `rocBLAS`, etc.
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/gentoo/gentoo/pull/26441#issuecomment-1203982128
> You are receiving this because you were mentioned.
> 
> Message ID: ***@***.***>
Theoretically it does not break rocBLAS and other high-level libraries,
since hip does the abstraction. Nevertheless I'll test it.
--

**Comment by heroxbd** [comment]
littlewu2508 ***@***.***> writes:

>> as a dependence of `hip`, should `roctracer` be bumped together?
>> 
>> Message ID: ***@***.***>
> dev-util/hip-5.1.3 does not depend on roctracer. It just use roctracer's
> header so it's just downloaded as another distfile

roctracer depends on `hip:${SLOT}` with exactly the same SLOT. If that's
the case, roctracer should be bumped together with hip as the "low level
libraries".  Otherwise users face slot conflicts.

**Comment by littlewu2508** [comment]
> as a dependence of `hip`, should `roctracer` be bumped together?

I have bumped roctracer and rocprofiler here. However, merge https://github.com/gentoo/gentoo/pull/26755 first because there are important fixes

**Comment by heroxbd** [comment]
Why commit 95d87d8c8860263b610760dc800e0599f2f75a03 is closing https://bugs.gentoo.org/857126?  They look completely unrelated.

**Comment by littlewu2508** [comment]
On Wed, Aug 03, 2022 at 06:51:20AM -0700, Benda Xu wrote:
> > as a dependence of `hip`, should `roctracer` be bumped together?
> 
> If you are merging `roctracer` and `rccl` into `hip`, add blocking dependence to `hip`, like `!dev-util/roctracer`.
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/gentoo/gentoo/pull/26441#issuecomment-1203979034
> You are receiving this because you were mentioned.
> 
> Message ID: ***@***.***>
I don't understand.

For roctracer, the bundling is only for the header. There is still need
for dev-util/roctracer.

For rccl, maybe you mean rocclr? So it hip-5.1.3 should block rocclr
--
